python/ray/tune/utils/placement_groups.py

```diff
@@ -429,18 +463,18 @@ def get_full_actor_cls(self, trial: "Trial",
         self._in_use_trials[trial] = pg
 
         # We still have to pass resource specs
-        # Pass the full resource specs of the first bundle per default
-        first_bundle = pg.bundle_specs[0].copy()
-        num_cpus = first_bundle.pop("CPU", None)
-        num_gpus = first_bundle.pop("GPU", None)
-
-        # Only custom resources remain in `first_bundle`
-        resources = first_bundle or None
-
-        if num_cpus is None:
-            # If the placement group specifically set the number
-            # of CPUs to 0, use this.
-            num_cpus = pgf.head_cpus
+        if not pgf.head_bundle_is_empty:
+            # Pass the full resource specs of the first bundle per default
+            head_bundle = pg.bundle_specs[0].copy()
+            num_cpus = head_bundle.pop("CPU", 0)
+            num_gpus = head_bundle.pop("GPU", None)
+
+            # Only custom resources remain in `head_bundle`
+            resources = head_bundle
+        else:
+            num_cpus = None
+            num_gpus = None
+            resources = None
```

Comment:
Does this work? Do we have a test for this path?

Reply:
@rkooo567 This is the case we were discussing today. Basically actor itself doesn't need anything. But its child will require some non trivial resources. Please advice what is the right way to call PlacementGroup in this case.

Reply:
https://sourcegraph.com/github.com/ray-project/ray/-/blob/python/ray/tune/integration/horovod.py?L256
The previous behavior is using legacy resources, which we are trying to remove here.

So it seems like pg and resources are not reaching feature parity?

Reply:
If it is not too much headache Core side, I would propose that we should reach feature parity by allowing actor using 0 resources.

Reply:
oh actually it reaches the feature parity. You can schedule actor/task with 0 resource requests. You just cannot create 0 resource bundle (which doesn't make sense).